src/librustc_resolve/lib.rs

```diff
@@ -753,6 +753,9 @@ pub struct ModuleS<'a> {
     parent_link: ParentLink<'a>,
     def: Option<Def>,
 
+    // The node id of the closest normal module (`mod`) ancestor (including this module).
```

Comment:
> So, for example for an item defined inside a function inside a module foo, this would be the id for foo?

Yeah. It is used for evaluating `self` and `super` in paths, resolving `pub(restricted)` to a `ty::Visibility`, more efficient implementation of `NodeIdTree`, and (most importantly) checking the visibility of imports and paths in the module (c.f. [`is_accessible`](https://github.com/rust-lang/rust/pull/35894/files#diff-aeb0880081a991f34aef2ab889e1fb7aL3278)).